dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/edit-payment/edit-payment.event-handler.ts

```diff
@@ -0,0 +1,49 @@
+import { EntityManager } from 'typeorm';
+import { DbRequestSource, FeeRecordEntity, PaymentEntity, UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { BaseUtilisationReportEvent } from '../../event/base-utilisation-report.event';
+import { FeeRecordStateMachine } from '../../../fee-record/fee-record.state-machine';
+import { feeRecordsMatchAttachedPayments } from '../helpers';
+
+type EditPayloadEventPayload = {
+  transactionEntityManager: EntityManager;
+  payment: PaymentEntity;
+  feeRecords: FeeRecordEntity[];
+  paymentAmount: number;
+  datePaymentReceived: Date;
+  paymentReference: string | undefined;
+  requestSource: DbRequestSource;
+};
+
+export type UtilisationReportEditPaymentEvent = BaseUtilisationReportEvent<'EDIT_PAYMENT', EditPayloadEventPayload>;
+
+export const handleUtilisationReportEditPaymentEvent = async (
+  report: UtilisationReportEntity,
+  { transactionEntityManager, payment, feeRecords, paymentAmount, datePaymentReceived, paymentReference, requestSource }: EditPayloadEventPayload,
+): Promise<UtilisationReportEntity> => {
+  payment.update({
+    amount: paymentAmount,
+    dateReceived: datePaymentReceived,
+    reference: paymentReference,
+    requestSource,
+  });
+  await transactionEntityManager.save(PaymentEntity, payment);
+
+  const feeRecordsAndPaymentsMatch = await feeRecordsMatchAttachedPayments(feeRecords, transactionEntityManager);
```

Comment:
Have we tested manually that these fee records have the new payment amount value and not a reference to the old value? (I remember having to be careful with deleting to fetch the array of payments again after deleting, although this could be different as is a value changing not a full payment no longer existing!)

Reply:
I feel like I might be missing something because I also don't see how the fee records have attached payments or where we are calling the new repo method that has been added to the utilisation report repo...

Reply:
I had tested that this works correctly locally, but you are right that it could do with some test coverage in the api test (and in e2e tests in an upcoming PR). The `feeRecordsMatchAttachedPayments` function loads the attached payments when it is called (hence the `await`), so the payments it checks against are the up to date once

Reply:
Ahh nice, I failed to realise that helper was fetching the payments fresh and not just using the ones currently attached. That makes sense!